ratelimit/leakybucket.go

```diff
@@ -0,0 +1,118 @@
+package ratelimit
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+
+	"github.com/zalando/skipper/metrics"
+	"github.com/zalando/skipper/net"
+)
+
+// The leaky bucket is an algorithm based on an analogy of how a bucket with a constant leak will overflow if either
```

Comment:
Godoc starts with the name of the type:
// ClusterLeakyBucket ...

This pattern is in Godoc for everything besides package docs, which starts with 
// Package <package-name> ...

Reply:
Moved this doc to the constructor method